### ISSUE TITLE
feature: flag to show raw combinations / traditional keyspace

### ISSUE THREAD

**Issue by roycewilliams** [hashcat/hashcat#3231]
**Describe the feature**

A flag to show the "raw" / traditional number of candidates for a given attack.

**Reasoning**

User confusion / FAQ activity around the existing `--keyspace` flag appears to be due to a combination of factors:

* The current meaning of the `--keyspace` flag in hashcat itself is different from colloquial use of the word 'keyspace'
* The colloquial use of the word 'keyspace' corresponds with a concept that users actually find useful when learning about and working with hash cracking

I believe that this means that there is a "ghost" / "missing" feature revealed by this confusion.

The requested feature would fill this gap, as an advisory / "nice to have" flag with new functionality. It would calculate the total number of candidates assembled for the attack, and then exit.

This calculated value is different from other current metrics around expected workload - different from hashcat's "work" keyspace value, different from the projected wallclock time, independent from the number of target hashes, etc. The value is intended to align with "back of the napkin" complexity analysis -  a convenience value used to perform rough work estimation during manual cracking workflows, or to inform discussion about theoretical attack complexity across platforms.

**Examples**

* For -a 0 wordlist.txt, it would simply show the number of words in the input wordlist.
* For -a 0 wordlist -r rules1.rule, it would show the number of words multiplied by the number of *valid* rules in rules1.rule.
* If rules files were stacked, it would multiply their lengths together.
* For -a 1 wordlist1.txt wordlist2.txt, it would multiply the line counts of each file.
* For -a 3, it would be equivalent to --combinations in maskprocessor.
etc.

**Naming**

If this were a 'green field' feature request, I would advocate for the existing hashcat-specific`--keyspace` functionality to be moved to its own more specialized flag, something like `--work-keyspace` ... but this would be a breaking change for much automation. So I suspect we would have to leave the existing flag as is. I'm open to suggestions about what to call this new feature. Perhaps `--raw-keyspace` or `--combinations`?

**Command line parsing considerations**

For obvious reasons, this flag would not be permitted to be combined with certain other flags, such as `--stdin`.

**Comment by Chick3nman** [comment]
Looking back at this now and I think implementing this as `--candidates` makes the most sense to help differentiate from hashcat's usage of "keyspace". Thoughts?

**Comment by roycewilliams** [comment]
For people coming to it, cold, the word "candidates"  all by itself doesn't really imply a count or total. But I can't immediately think of anything that's an obvious win. "--raw-keyspace" is the most descriptive, but is a bit wordy.

**Comment by Chick3nman** [comment]
I could see that but in finding out what `--candidates` is, the user will learn the terminology for what a "Candidate" is in context, which we use elsewhere. Trying to separate terminology a bit in context.

**Comment by roycewilliams** [comment]
I think that sets up a dynamic that has bitten us in the past - where a naive user would _reasonably_ think that `--candidates` would do something like _show_ all attempted candidates, or change the display of candidates in some circumstances. It's just way too vague for my tastes. In the same vein, I feel that `--show` is insufficiently precise in retrospect, and should have been given a more precise name. I think `--candidates` is significantly insufficiently descriptive.

**Comment by Chick3nman** [comment]
That's fair. With `--show` there was precedent in John that we were following, at least as I understand it. perhaps `--total-candidates`?

**Comment by roycewilliams** [comment]
that's a nice compromise - I like it!
